Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
Actually, I think I have got your idea. I think it is probably might be better to remove projectiles in `CheckPassable` functions (because sometimes `CheckPassable` might be called outside of `Projectile::Process()`). But in this case you should do something like that:
```
bool Projectile::CheckPassable()
{
    if (IMovable::CheckPassable())
    {
        return true;
    }
    Delete();
    return false;
}
```
So when projectile will not be able to move it will be deleted. I think it is right approach. So you can keep `virtual` keyword -> and `Projectile::CheckPassable()` (but it will be needed to change `Process` & `CheckPassable`)